include/dlaf/matrix/hdf5.h

```diff
@@ -53,6 +53,32 @@ const H5::PredType& hdf5_datatype<std::complex<T>>::type = hdf5_datatype<T>::typ
 template <class T>
 struct hdf5_datatype<const T> : public hdf5_datatype<T> {};
 
+// Type to string mappings
+template<typename T>
+struct TypeToString {
+  static inline const std::string value = typeid(T).name();
+};
+
+template<>
+struct TypeToString<float> {
+  static inline const std::string value = "s";
+};
```

Comment:
Not worth much bikeshedding as long as this stays internal. For reference:

- We use this for miniapps: https://github.com/eth-cscs/DLA-Future/blob/08d13255892e6c3e6e7023434770c6db1f788543/miniapp/include/dlaf/miniapp/options.h#L114-L132. It's similar, but not exactly the same, and at some point unifying it may be useful.

This should include `typeinfo`. On top of that I'd probably do:
```suggestion
#include <string_view>

// Type to string mappings
template<typename T>
struct TypeToString {
  static inline constexpr std::string_view value = typeid(T).name();
};

template<>
struct TypeToString<float> {
  static inline constexpr std::string_view value = "s";
};
```

and later:
```
template <typename T>
inline constexpr std::string_view TypeToString_v = TypeToString<T>::value;
```
to have a shorthand for it.

Reply:
Apart from the `ElementType` information, I think I addressed all the comments. Thank you for the feedback.